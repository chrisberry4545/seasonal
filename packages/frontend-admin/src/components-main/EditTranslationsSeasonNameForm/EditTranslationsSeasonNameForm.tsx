import React, { FC } from 'react';
import {
  GetAuthorizedBackendData
} from '../GetAuthorizedBackendData/GetAuthorizedBackendData';
import { useParams } from 'react-router-dom';
import {
  getOneTranslationsSeasonName,
  updateTranslationsSeasonName
} from '../../services';
import {
  BaseFormTranslationsSeasonName
} from '../BaseFormTranslationsSeasonName/BaseFormTranslationsSeasonName';
import { LayoutWithTitle } from '../../components-layouts';

export const EditTranslationsSeasonNameForm: FC<{}> = () => {
  const { id } = useParams();
  return (
    <LayoutWithTitle title='Edit Translations Season Name'>
      <GetAuthorizedBackendData
        InnerComponent={BaseFormTranslationsSeasonName}
        requestDataMethod={() => getOneTranslationsSeasonName(id as string)}
        updateMethod={updateTranslationsSeasonName}
        />
    </LayoutWithTitle>
  );
};
