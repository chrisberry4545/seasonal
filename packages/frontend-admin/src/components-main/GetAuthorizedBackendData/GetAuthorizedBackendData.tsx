import React, { FC, useState, useEffect } from 'react';

export interface IGetAuthorizedBackendDataProps<T> {
  items: T;
  reload?: () => void;
  updateMethod?: (item: T) => Promise<T>;
}

interface IGetAuthorizedBackendDataState<T> {
  isLoading: boolean;
  items: T | null;
  error: string | null;
}

export function GetAuthorizedBackendData<T>(
  InnerComponent: FC<IGetAuthorizedBackendDataProps<T>>,
  requestDataMethod: () => Promise<T>,
  updateMethod?: (item: T) => Promise<T>
): FC<{}> {
  return () => {
    const [state, setState] = useState<IGetAuthorizedBackendDataState<T>>({
      error: null,
      isLoading: true,
      items: null
    });

    const requestNewData = () => {
      requestDataMethod()
        .then((items) => setState({
          error: null,
          isLoading: false,
          items
        })).catch((error) => setState({
          error: error.message,
          isLoading: false,
          items: null
        }));
    };

    useEffect(requestNewData, []);

    return (
      <div>
        {
          state.isLoading
            ? <div>Loading...</div>
            : <div>
              {
                !state.error && state.items
                  ? <InnerComponent
                      items={state.items}
                      reload={requestNewData}
                      updateMethod={updateMethod}
                    />
                  : <div>{state.error}</div>
              }
            </div>
        }
      </div>
    );
  };
}
