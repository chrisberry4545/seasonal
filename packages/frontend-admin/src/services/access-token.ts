import { getStorage } from '../consts';

export const setAccessToken = (token: string) => {
  getStorage().accessToken = token;
};

export const getAccessTokenHeaders = (): HeadersInit => {
  const token = getStorage().accessToken;
  return token
  ? {
    Authorization: `Bearer ${token}`
  }
  : {};
};
