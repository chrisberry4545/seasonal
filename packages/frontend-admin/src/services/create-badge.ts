import { IBadge } from '@chrisb-dev/seasonal-shared-models';
import { makeAuthorizedRequest } from './make-authorized-request';
import { BADGE_URL } from '../config';

export const createBadge = async (badge: IBadge): Promise<IBadge> =>
  makeAuthorizedRequest<IBadge>(BADGE_URL, {
    body: JSON.stringify(badge),
    method: 'POST'
  });
