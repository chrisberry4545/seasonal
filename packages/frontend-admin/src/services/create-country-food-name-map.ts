import { ICountryFoodNameMap } from '@chrisb-dev/seasonal-shared';
import { makeAuthorizedRequest } from './make-authorized-request';
import { COUNTRY_FOOD_NAME_MAP_URL } from '../config';

export const createCountryFoodNameMap = async (
  country: ICountryFoodNameMap
): Promise<ICountryFoodNameMap> =>
  makeAuthorizedRequest<ICountryFoodNameMap>(COUNTRY_FOOD_NAME_MAP_URL, {
    body: JSON.stringify(country),
    method: 'POST'
  });
