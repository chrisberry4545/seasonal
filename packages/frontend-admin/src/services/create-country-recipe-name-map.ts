import { ICountryRecipeNameMap } from '@chrisb-dev/seasonal-shared-models';
import { makeAuthorizedRequest } from './make-authorized-request';
import { COUNTRY_RECIPE_NAME_MAP_URL } from '../config';

export const createCountryRecipeNameMap = async (
  country: ICountryRecipeNameMap
): Promise<ICountryRecipeNameMap> =>
  makeAuthorizedRequest<ICountryRecipeNameMap>(COUNTRY_RECIPE_NAME_MAP_URL, {
    body: JSON.stringify(country),
    method: 'POST'
  });
