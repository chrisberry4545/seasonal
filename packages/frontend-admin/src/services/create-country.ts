import { ICountry } from '@chrisb-dev/seasonal-shared-models';
import { makeAuthorizedRequest } from './make-authorized-request';
import { COUNTRY_URL } from '../config';

export const createCountry = async (country: ICountry): Promise<ICountry> =>
  makeAuthorizedRequest<ICountry>(COUNTRY_URL, {
    body: JSON.stringify(country),
    method: 'POST'
  });
