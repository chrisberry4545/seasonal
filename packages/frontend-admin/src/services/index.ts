export * from './access-token';

export * from './get-all-countries';

export * from './get-all-food';

export * from './get-all-recipes';

export * from './get-single-recipe';

export * from './login-request';

export * from './make-authorized-request';

export * from './update-recipe';
