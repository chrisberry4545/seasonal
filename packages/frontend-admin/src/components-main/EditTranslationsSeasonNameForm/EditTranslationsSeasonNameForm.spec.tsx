
import React from 'react';
import { EditTranslationsSeasonNameForm } from './EditTranslationsSeasonNameForm';
import { shallow, ShallowWrapper } from 'enzyme';
import { ITranslationsSeasonName } from '@chrisb-dev/seasonal-shared-models';
import * as services from '../../services';
import { GetAuthorizedBackendData } from '../GetAuthorizedBackendData/GetAuthorizedBackendData';

jest.mock('../../services', () => ({
  getOneTranslationsSeasonName: jest.fn(),
  updateTranslationsSeasonName: jest.fn()
}));
jest.mock('../../components-layouts', () => ({
  LayoutWithTitle: () => 'LayoutWithTitle'
}));
jest.mock('../GetAuthorizedBackendData/GetAuthorizedBackendData', () => ({
  GetAuthorizedBackendData: () => 'GetAuthorizedBackendData'
}));
jest.mock('../BaseFormTranslationsCountryName/BaseFormTranslationsCountryName', () => ({
  BaseFormTranslationsCountryName: () => 'BaseFormTranslationsCountryName'
}));
jest.mock('react-router-dom', () => ({
  useParams: jest.fn().mockReturnValue({ id: 'id' })
}));

describe('<EditTranslationsSeasonNameForm />', () => {
  let wrapper: ShallowWrapper;
  let mockGetOne: jest.SpyInstance;
  let mockUpdate: jest.SpyInstance;
  const item = {
    name: 'name'
  } as ITranslationsSeasonName;

  beforeEach(() => {
    mockGetOne = jest.spyOn(services, 'getOneTranslationsSeasonName')
      .mockResolvedValue(item);
    mockUpdate = jest.spyOn(services, 'updateTranslationsSeasonName')
      .mockResolvedValue({} as ITranslationsSeasonName);
    wrapper = shallow(
      <EditTranslationsSeasonNameForm />
    );
  });

  test('renders correctly', () => expect(wrapper).toMatchSnapshot());

  test('can get the item data', async () => {
    await wrapper.find(GetAuthorizedBackendData).props().requestDataMethod();
    expect(mockGetOne).toHaveBeenCalledWith('id');
  });

  test('can update the item', () => {
    const update = wrapper.find(GetAuthorizedBackendData).props().updateMethod;
    if (update) {
      update(item);
    }
    expect(mockUpdate).toHaveBeenCalledWith(item);
  });

});
