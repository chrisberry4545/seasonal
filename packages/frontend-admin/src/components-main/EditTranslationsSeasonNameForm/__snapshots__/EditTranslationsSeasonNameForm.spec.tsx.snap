// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditTranslationsSeasonNameForm /> renders correctly 1`] = `
<LayoutWithTitle
  title="Edit Translations Season Name"
>
  <GetAuthorizedBackendData
    InnerComponent={[Function]}
    requestDataMethod={[Function]}
    updateMethod={[MockFunction]}
  />
</LayoutWithTitle>
`;
